Common/Securities/ContractSecurityFilterUniverse.cs

```diff
@@ -59,11 +63,45 @@ protected enum ContractExpirationType : int
         /// </summary>
         public DateTime LocalTime { get; private set; }
 
+        /// <summary>
+        /// All data in this filter
+        /// Marked internal for use by extensions
+        /// </summary>
+        /// <remarks>
+        /// Setting it will also set AllSymbols
+        /// </remarks>
+        internal IEnumerable<TData> Data
+        {
+            get
+            {
+                return _data;
+            }
+            set
+            {
+                _data = value;
+                _allSymbols = _data.Select(x => x.GetSymbol());
+            }
+        }
+
         /// <summary>
         /// All Symbols in this filter
         /// Marked internal for use by extensions
         /// </summary>
-        internal IEnumerable<Symbol> AllSymbols;
+        /// <remarks>
+        /// Setting it will remove any data that doesn't have a symbol in AllSymbols
+        /// </remarks>
+        internal IEnumerable<Symbol> AllSymbols
+        {
+            get
+            {
+                return _allSymbols;
+            }
+            set
+            {
+                _allSymbols = value;
+                _data = _data.Where(x => _allSymbols.Contains(x.GetSymbol()));
```

Comment:
Think maybe we should make symbols basically be a passthrough to data now, instead of having 2 sources of  truth here